pkg/podrescompat/client.go

```diff
@@ -0,0 +1,83 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podrescompat
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+
+	podresourcesapi "k8s.io/kubelet/pkg/apis/podresources/v1"
+
+	"github.com/openshift-kni/resource-topology-exporter/pkg/sysinfo"
+)
+
+type sysinfoClient struct {
+	sysInfoConfig string
+	cli           podresourcesapi.PodResourcesListerClient
+}
+
+func NewSysinfoClientFromLister(cli podresourcesapi.PodResourcesListerClient, sysInfoConfig string) podresourcesapi.PodResourcesListerClient {
+	return &sysinfoClient{
+		cli:           cli,
+		sysInfoConfig: sysInfoConfig,
+	}
+}
+
+func (sc *sysinfoClient) List(ctx context.Context, in *podresourcesapi.ListPodResourcesRequest, opts ...grpc.CallOption) (*podresourcesapi.ListPodResourcesResponse, error) {
+	return sc.cli.List(ctx, in, opts...)
+}
+
+func (sc *sysinfoClient) GetAllocatableResources(ctx context.Context, in *podresourcesapi.AllocatableResourcesRequest, opts ...grpc.CallOption) (*podresourcesapi.AllocatableResourcesResponse, error) {
+	resp, err := sc.cli.GetAllocatableResources(ctx, in, opts...)
+	if err != nil {
+		sysResp, sysErr := sc.makeAllocatableResourcesResponse()
```

Comment:
Maybe it's good to log the user that podresourceAPI was disabled by feature-gate and we're using some emulation instead.

Reply:
done.